rust/rope/src/rope.rs

```diff
@@ -478,7 +480,9 @@ impl Rope {
     /// Note: `edit` and `edit_str` may be merged, using traits.
     ///
     /// Time complexity: O(log n)
-    pub fn edit_str(&mut self, start: usize, end: usize, new: &str) {
+    pub fn edit_str<T>(&mut self, range: T, new: &str) where T: RangeBounds<usize> {
```

Comment:
Nit: I would consistently put the `where` clause on the next line.

Reply:
:+1: